### DISCUSSION TITLE
A look at the orderby LINQ keyword

### DISCUSSION THREAD

**Discussion by Rekkonnect** [dotnet/csharplang#5697] [Language Ideas] [answered]
`orderby` serves a single purpose in LINQ expressions, and it follows the syntax `orderby <expression> [<ordering>]` where `ordering` is etiher `ascending` or `descending`, defaulting to `ascending` if not explicitly provided. There is this other `by`-suffixed syntax, the `group <expression> by <expression>` syntax. The `by` is a keyword here.

Without knowing the exact purpose behind not using an already available keyword, `by`, in the `orderby` syntax, and creating a new keyword as a concatenation of the two keywords. Given `typeof` and `sizeof` (`nameof` wasn't there at that time), it makes some sense to merge the *preposition* like already done. But, in this instance, even though forced, `by` is a separate keyword on its own.

It's not *too* late to introduce a new way to express `orderby`, like so: `order by`. `order` will be a newly introduced LINQ keyword, and `by` will be the same keyword as before, except it can also come after an `order` keyword.

The massive change would be the following example code to see the occurrent change:
```csharp
orderby book.Copies descending
order by book.Copies descending
```

In a more detailed example:
```csharp
from book in books
where book.ReleaseDate > 2015 and <= 2020
orderby book.Copies descending
select book;
```
could also be written as:
```csharp
from book in books
where book.ReleaseDate > 2015 and <= 2020
order by book.Copies descending
select book;
```

Side-effects: Extra effort must be made to not explode the parser from `group order by ...`. It would not be too weird for a variable to be called `order`, but thankfully this change does not make the syntax ambiguous.

Real side-effects: Another reason to divide programmers on their styling opinions. `orderby` will continue being legal for obvious backwards-compatibility reasons, and "the battle of the true order by style" will begin.

### Useful FAQ

**Is this a concerning matter?**
No.

**Will the world come to an end because of this?**
No.

**Do people lose their sanity writing this?**
No.

**Does it bug you more than once in a blue moon, unless you're writing specialized code?**
No.

**Then why are you making it a deal?**
It's a discussion, thought I'd drop a few cents.

**Comment by HaloFour** [comment]
I'd suggest that the syntax has already been decided here and having two ways of doing things for the sake of style wouldn't be preferable.

**Comment by TahirAhmadov** [comment]
"Order by" and "Group by" in SQL make no sense to be 2 separate words. Which is why in LINQ, "orderby" is one word. "Group xxx by yyy" is 2 words because of 2 expressions; it's not a styling inconsistency, rather a syntax necessity.

**Comment by CyrusNajmabadi** [comment, marked as answer]
We explicitly considered this when designing linq and rejected it.  It's just extra typing for the user for no benefit as far as we could see.
